docs/tutorials/initializedirtyimage.py

```diff
@@ -0,0 +1,190 @@
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.11.2
+#   kernelspec:
+#     display_name: Python 3
+#     language: python
+#     name: python3
+# ---
+
+# + nbsphinx="hidden"
+# %matplotlib inline
+
+# + nbsphinx="hidden"
+# %run notebook_setup
+# -
+
+# # Not completely done, still have to do some links do some fleshing out, just want to get it here to make sure there isn't anything critically wrong.
+#
+# ## Initializing Model with the Dirty Image
+#
+# In this tutorial we will be going over how to optimize the model based off of the dirty image. It has a low loss value (maximal likelihood) and so this allows the optimization to be much quicker. The problem with the dirty image is that certain model paramterizations enforce image positivity while the dirty image usually contains regions of negative flux. So in this tutorial we will be optimizing the model to be similar to the dirty image (phrasing could be better here). We will also be going over saving and loading the model. After saving and loading it, it can be then optimized against the visibility data to complete it (though this will not be done in this tutorial). We will be using the dataset of the ALMA logo first used in the Gridding and Diagnostic Images tutorial.
+#
+#
+
+# ### Image Setup
+#
+# Here we will be setting up the ALMA Logo image dataset and displaying it. Consult the Gridding and Diagnostic Images Tutorial for reference.
+
+import torch
+import numpy as np
+import matplotlib.pyplot as plt
+from mpol import gridding, coordinates, precomposed, losses, utils
+from astropy.utils.data import download_file
+from torch.utils.tensorboard import SummaryWriter
+
+# In this tutorial we will be saving and loading the model we use and it is important to note that when saving and loading a model the cell_size, nchan, and npix are the same between the model and the coordinates if used in another document.
+
+# +
+# load the mock dataset of the ALMA logo
+fname = download_file(
+    "https://zenodo.org/record/4498439/files/logo_cube.npz",
+    cache=True,
+    show_progress=True,
+    pkgname="mpol",
+)
+
+# this is a multi-channel dataset... but for demonstration purposes we'll use
+# only the central, single channel
+chan = 4
+d = np.load(fname)
+uu = d["uu"][chan]
+vv = d["vv"][chan]
+weight = d["weight"][chan]
+data_re = d["data_re"][chan]
+data_im = -d["data_im"][
+    chan
+]  # we're converting from CASA convention to regular TMS convention by complex conjugating the visibilities
+
+# define the image dimensions, making sure they are big enough to fit all
+# of the expected emission
+coords = coordinates.GridCoords(
+    cell_size=0.03, npix=180
+)  # The smaller the cell size and the larger the npix can greatly increase run time
+gridder = gridding.Gridder(
+    coords=coords,
+    uu=uu,
+    vv=vv,
+    weight=weight,
+    data_re=data_re,
+    data_im=data_im,
+)
+
+# export to PyTorch dataset
```

Comment:
I cloned your fork and tried to build the docs on your branch, but got an error here. Your tutorial is written for the new syntax that Brianna created (which is good), but I think your fork still needs to incorporate the upstream changes from MPoL-dev/main. I think you can do this easily using the "Fetch Upstream" changes button on the main page of your forked repo. I'm kind of confused how this builds correctly for you locally when you run `make html`? It seems like it builds correctly in the pull request because those changes are being applied on top of the new syntax changes already in MPoL-dev/main.

Reply:
I made sure I had done fetch origin and everything before doing the pull request (or at least I thought I did) and when I do make html the documentation was all building properly for me. I'll try and see what's up with that.

Reply:
I think fetch origin will just update your local repository with the changes to RCF42/MPoL, which should be none, since you were the only one making changes. What you want to do is update RCF42/MPoL with the new changes that are in MPoL-dev/MPoL. I think you can do this using the "Fetch Upstream" button I mentioned. See also here: https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/merging-an-upstream-repository-into-your-fork

Reply:
Ah okay, my bad

Reply:
So I've done the fetch upstream and made sure that my branch is up to date with the overall main but I'm facing a problem that when I try and build the documentation it will make just about everything fine but when it comes to the tutorials it gives me:
 `E:\robert\Documents\GitHub\MPoL\docs\index.rst:33: WARNING: toctree contains reference to nonexisting document 'tutorials/gridder'`
for every single tutorial and nothing relating tutorials seems to appear under the build folder. Even if I run it off of my main branch (which is even with this one's) the problem still occurs.

Reply:
Is it possible this is a local pip install issue, as per this [discussion](https://github.com/MPoL-dev/MPoL/discussions/66#discussioncomment-814423)?

Reply:
I don't think so because with pip list MPoL appears the way you said it should and both Sphinx and jupytext are also there. I have checked my local branch and I also do have that commit and gridded visibilities is _grid_visibilities. I do get some other errors when doing make html (like it says [Errno 2] No such file or directory: '_static/baselines/build/baselines.csv'.) but none of them appear the error about the 'nonexistent' tutorials.

Reply:
What happens if you do

```
$ python
>>> import mpol 
>>> mpol.__version__
>>> from mpol.gridding import Gridder
>>> dir(Gridder)
```
?

This should show the updated routines, like 

```['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_fliplr_cube', '_grid_visibilities', '_grid_weights', '_histogram_channel', '_histogram_cube', '_null_dirty_beam', 'get_dirty_beam', 'get_dirty_beam_area', 'get_dirty_image', 'sky_vis_gridded', 'to_pytorch_dataset']```

Also, are there any build errors that happen before the one you listed?

You might also try doing a ``make clean`` before you restart ``make html``, just in case stale files are messing something up.

Reply:
All the updated routines appear and there are no build errors that happen before the one I listed. I have been doing the make clean to make html as well. I've also done fresh installs of it all and still get the same problem. I'll see what happens when I switch what computer I'm working on, hopefully that will work.

Reply:
I've tried it completely fresh from a new computer and it still didn't work so I'm mildly at a lost for why the problem's happening.

Reply:
In your directory and virtual machine if you do the equivalent of "pip show jinja2" for your system, do you get version 3.0.1 or version 2.11?

Reply:
After talking with Robert, I think his main issue stems from how `make` works (or doesn't) on windows. But I did notice this error when I was installing a fork recently. The strange thing is that it appears to be transient... I think it's nbsphinx that relies upon jinja2 2.11, but sometimes the latest jinja2 ends up being installed by accident. I'm not really sure how this is possible with pip, that the dependencies are resolved non-deterministically. @hgrzy have you come across this error too, and if so, was it semi-random?

Reply:
Yes, I was getting that error a week or two ago, but I just tried replicating it and it seems to install the jinja2 version that is desired even though in both cases I was using pip version 20.2.3. I saved the terminal as a text file when I was getting the error though and it always occurred in the the process of installing the developer package at the point where it's running the setup.py files, though it isn't any particular setup.py file. It also included this in the output:

`ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.`

Reply:
I just looked at the release history for nbsphinx. As of 10 hours ago pip install nbsphinx now installs version 0.8.6 (which seems to run jinja2 version 3.0.1 fine) instead of 0.8.5 (requires jinja2 version 2.11.0) which is probably why I cannot replicate the error. So maybe this is no longer an issue?

Reply:
Good detective work! Hopefully this is resolved for future builds---it seems specifically an nbsphinx problem. If it crops up again we'll need to pin the jinja2 version, which I'd like to avoid doing unnecessarily.